src/notification.coffee

```diff
@@ -0,0 +1,45 @@
+superagent = require 'superagent'
+
+class Notification
+  constructor: (options={}) ->
+    for key in Notification.REQUIRED_KEYS
+      unless options.hasOwnProperty(key)
+        throw new Error "Required key missing: `#{key}`"
+
+      @[key] = options[key]
+
+    for key in Notification.OPTIONAL_KEYS
+      keyAllowed = options.hasOwnProperty(key) &&
+        (-1 == Notification.SERVERSIDE_KEYS.indexOf(key))
+
+      if keyAllowed
+        @[key] = options[key]
+
+  # TODO:
+  # This should probably use restify.JsonClient() or ganomede client lib,
+  # but let's just keep things simple for now.
+  send: (uri, callback) ->
```

Comment:
Yes, should use restify instead of superagent.

One main reason for that, is that restify adds dtrace support (to debug latency issues), superagent doesn't have that, so we may have "holes" in the dtrace reports.

I will make an issue.